lms/djangoapps/certificates/migrations/0025_cleanup_certificate_errors.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 2.2.20 on 2021-04-27 18:12
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    """
+    Clean records in the `CERTIFICATES_GENERATEDCERTIFICATE` table that are in the downloadable state but also have
+    old errors still part of their certificate record.
+
+    As part of this migration we are also altering the Managers for the GeneratedCertificate model. We need the ability
+    to access the `objects` attribute for the cleanup.
+    """
+    def cleanup_certificate_records(apps, schema_editor):
+        GeneratedCertificate = apps.get_model('certificates', 'GeneratedCertificate')
+
+        cert_list = (
+            GeneratedCertificate.objects
+            .filter(status='downloadable')
+            .exclude(error_reason='')
+        )
+
+        for cert in cert_list:
+            cert.error_reason = ''
+            cert.save()
```

Comment:
I don't know how many objects this will match in prod, but it'll be a lot faster to migrate if you just use `update` instead. E.g.
```
GeneratedCertificate.objects.filter(status='downloadable').exclude(error_reason='').update(error_reason='')
```
See: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.update

Note: `.update()` runs a SQL update (and thus doesn't use the `save()` method which causes it to not show up in the history table. I think that's fine here, but if you're expecting `save()` to handle anything else for you, it won't. :)

Reply:
Thanks! That sounds promising. I'll plan on making that change shortly.